data.py

```diff
@@ -389,18 +404,21 @@ def __init__(
         self.source_folders = self.interferer_dirs + [self.target_dir]
         self.source_tracks = self.get_tracks()
         self.nb_samples = nb_samples
+        self.random_tracks = random_tracks
+        self.seed = seed
+        random.seed(self.seed)
 
     def __getitem__(self, index):
-        # for validation, get deterministic behavior
-        # by using the index as seed
-        if self.split == 'valid':
-            random.seed(index)
-
         # For each source draw a random sound and mix them together
         audio_sources = []
         for source in self.source_folders:
             # select a random track for each source
+            if self.random_tracks:
```

Comment:
This will ensure that we have a deterministic behavior for the random track selection, right?

I think it is important to note in the documentation that the random number generator is seeded and that in one epoch we should have more samples than there are `source_tracks` such that we have more random mixtures.

Maybe it is better to not set the seed here?

Reply:
> Maybe it is better to not set the seed here?

I think this is the only way to have deterministic behaviour since we do not have access to the `epoch end` flag (we can redo this once this is is converted to pytorch-lightning). What we want is that for validation each epoch yield example the same samples without being limited to just have one sample per track. Imagine a dataset of many speech files with one large 2h long noise recording that we want to yield mixtures from.

> I think it is important to note in the documentation that the random number generator is seeded and that in one epoch we should have more samples than there are source_tracks such that we have more random mixtures.

not sure if I get you right here? We should make it more clear that we draw multiple samples per track even for validation?

Reply:
I was just thinking about the case that we have 100 speech files + 100 noise files and the user is using 100 samples in one batch. Now, in each epoch he will basically train on the same 100 mixtures although he might assume that `self.random_tracks = True` will give him 100 different tracks in each epoch.

But I have seen that you have removed the `self.random_tracks`. Now everything looks fine 👍

Reply:
understood! okay that is indeed confusing...Thanks!